### DISCUSSION TITLE
Readonly cli

### DISCUSSION THREAD

**Discussion by macbryla** [duckdb/duckdb#9098] [Q&A] [answered]
I have a rust application reading and writing to duckdb. 
When I open duckdb using CLI utility using 
`duckdb -readonly energy.db`
my rust code fails when trying to insert a record
`IO Error: Could not set lock on file "data/energy.db": Resource temporarily unavailable`

I am clearly missing something here, so I am hoping someone can shine some light on this.
I was under the impression that in the above example CLI connection would be read-only (just for that connection), and my app would have access to a read-write connection. 

The use case is to have a read-only access from CLI, while the app runs short-lived read-write and read-only sessions.

Thank you.

**Comment by ned2** [comment, marked as answer]
When you open writeable connection to a DuckDB disk-based database, the database file is locked from connections from other processes. So DuckDD only supports concurrent connections across different processes if they are all read-only. So you'll need to first close any write-enabled connections before other processes can read from the database.

See this entry in the FAQ: https://duckdb.org/faq#how-does-duckdb-handle-concurrency

**Reply by macbryla** [reply]
Thank you. 
My confusion was around the multi-process access. Now, it makes sense.
